application_sdk/server/fastapi/__init__.py

```diff
@@ -148,6 +154,37 @@ def __init__(
         # Initialize parent class
         super().__init__(handler)
 
+    def _is_duckdb_ui_running(self, host="0.0.0.0", port=4213):
+        """Check if DuckDB UI is already running on the default port."""
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.settimeout(0.5)
+            result = sock.connect_ex((host, port))
+            return result == 0
+
+    def _start_duckdb_ui(self, db_path="/tmp/logs/observability.db"):
+        """Start DuckDB UI if not already running, and attach the /tmp/logs folder."""
+        logs_dir = "/tmp/logs"
+        if not self._is_duckdb_ui_running():
+            os.makedirs(logs_dir, exist_ok=True)
+            con = duckdb.connect(db_path)
+            # Attach all .parquet files in /tmp/logs as tables
+            for fname in os.listdir(logs_dir):
+                fpath = os.path.join(logs_dir, fname)
+                if fname.endswith(".parquet"):
+                    tbl = os.path.splitext(fname)[0]
+                    con.execute(
+                        f"CREATE OR REPLACE VIEW {tbl} AS SELECT * FROM read_parquet('{fpath}')"
+                    )
+            # Start DuckDB UI using SQL command
+            con.execute("CALL start_ui();")
+            self._duckdb_ui_con = con  # Store the connection, do NOT close it!
+
+    def observability(self, request: Request) -> RedirectResponse:
+        """Endpoint to launch DuckDB UI for log self-serve exploration."""
+        self._start_duckdb_ui()
+        # Redirect to the local DuckDB UI
+        return RedirectResponse(url="http://0.0.0.0:4213")
```

Comment:
can we move the duckdb ui into a separate file or module?

ideally observability (which captures all things logs, traces, metrics etc)? - let's check what other SDKs do in such cases.